jbmc/src/java_bytecode/remove_instanceof.cpp

```diff
@@ -110,70 +110,31 @@ bool remove_instanceoft::lower_instanceof(
       return a.compare(b) < 0;
     });
 
-  // Make temporaries to store the class identifier (avoids repeated derefs) and
-  // the instanceof result:
-
   auto jlo = to_struct_tag_type(java_lang_object_type().subtype());
   exprt object_clsid = get_class_identifier_field(check_ptr, jlo, ns);
 
-  symbolt &clsid_tmp_sym = fresh_java_symbol(
-    object_clsid.type(),
-    "class_identifier_tmp",
-    this_inst->source_location,
-    function_identifier,
-    symbol_table);
-
-  symbolt &instanceof_result_sym = fresh_java_symbol(
-    bool_typet(),
-    "instanceof_result_tmp",
-    this_inst->source_location,
-    function_identifier,
-    symbol_table);
-
-  // Create
-  // if(expr == null)
-  //   instanceof_result = false;
-  // else
-  //   string clsid = expr->@class_identifier
-  //   instanceof_result = clsid == "A" || clsid == "B" || ...
-
-  // According to the Java specification, null instanceof T is false for all
-  // possible values of T.
-  // (http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2)
-
-  code_blockt else_block;
-  else_block.add(code_declt(clsid_tmp_sym.symbol_expr()));
-  else_block.add(code_assignt(clsid_tmp_sym.symbol_expr(), object_clsid));
+  // Replace the instanceof operation with (check_ptr != null &&
+  //  (check_ptr->@class_identifier == "A" ||
+  //   check_ptr->@class_identifier == "B" || ...
+
+  // Note we deliberately do not use a temporary for the *check_ptr redundant
+  // subexpression because by testing check_ptr->@class_identifier rather than
+  // (e.g.) tmp = check_ptr->@class_identifier followed by testing 'tmp', we
+  // increase the chances that goto-symex can reduce the alias set of check_ptr
+  // according to the condition imposed here.
```

Comment:
That's all very well explained, but not as complete as the commit message. I don't think anyone would complain if you pasted it here completely, thus making the life of future maintainers ever so wonderful.